apps/api/src/pages/pages.service.ts

```diff
@@ -0,0 +1,767 @@
+import * as Y from 'yjs';
+import { Injectable, Logger, Inject } from '@nestjs/common';
+import { PrismaService } from '@/common/prisma.service';
+import { MiscService } from '@/misc/misc.service';
+import { CanvasService } from '@/canvas/canvas.service';
+import { KnowledgeService } from '@/knowledge/knowledge.service';
+import { MinioService } from '@/common/minio.service';
+import { MINIO_INTERNAL } from '@/common/minio.service';
+import { streamToBuffer } from '@/utils';
+import { User, EntityType } from '@refly-packages/openapi-schema';
+import { ShareNotFoundError } from '@refly-packages/errors';
+import { CodeArtifactService } from '@/code-artifact/code-artifact.service';
+import { ShareService } from '@/share/share.service';
+import {
+  UpdatePageDto,
+  PageDetailResult,
+  UpdatePageResult,
+  SharePageResult,
+  DeletePageResult,
+  DeletePageNodeResult,
+  Page,
+  PageNodeRelation,
+} from './pages.dto';
+import { createId } from '@paralleldrive/cuid2';
+
+// Type definitions
+interface ResolveUserResponse {
+  uid: string;
+  userInfo?: {
+    name?: string;
+  };
+}
+
+// Custom error class for page not found
+class PageNotFoundError extends ShareNotFoundError {
+  code = 'E1010'; // Custom error code
+  messageDict = {
+    en: 'Page not found',
+    'zh-CN': 'Page not found', // Changed to English
+  };
+}
+
+// Generate unique IDs
+const genPageId = (): string => `page-${createId()}`;
+const genPageNodeRelationId = (): string => `pnr-${createId()}`;
+
+@Injectable()
+export class PagesService {
+  private logger = new Logger(PagesService.name);
+
+  constructor(
+    private prisma: PrismaService,
+    private miscService: MiscService,
+    private canvasService: CanvasService,
+    private knowledgeService: KnowledgeService,
+    private codeArtifactService: CodeArtifactService,
+    private shareService: ShareService,
+    @Inject(MINIO_INTERNAL) private minio: MinioService,
+  ) {}
+
+  // List all pages for a user
+  async listPages(
+    user: User,
+    page = 1,
+    pageSize = 20,
+  ): Promise<{
+    pages: Page[];
+    total: number;
+    page: number;
+    pageSize: number;
+  }> {
+    const pages = await this.prisma.$queryRaw<Page[]>`
```

Comment:
`$queryRaw` is error prone, use generated query methods instead, like `this.prisma.pages.findMany`